src/common/hist_util.cc

```diff
@@ -252,13 +257,52 @@ void RowsWiseBuildHistKernel(Span<GradientPair const> gpair, Span<bst_idx_t cons
 
     // The trick with pgh_t buffer helps the compiler to generate faster binary.
     const float pgh_t[] = {p_gpair[idx_gh], p_gpair[idx_gh + 1]};
-    for (size_t j = 0; j < row_size; ++j) {
-      const uint32_t idx_bin =
-          two * (static_cast<uint32_t>(gr_index_local[j]) + (kAnyMissing ? 0 : offsets[j]));
-      auto hist_local = hist_data + idx_bin;
-      *(hist_local) += pgh_t[0];
-      *(hist_local + 1) += pgh_t[1];
-    }
+    #if defined(SVE_SUPPORT_DETECTED)
+      svfloat64_t pgh_t0_vec = svdup_n_f64(pgh_t[0]);
+      svfloat64_t pgh_t1_vec = svdup_n_f64(pgh_t[1]);
```

Comment:
It seems you don't need the `pgh_t` in the SVE code section. Could you please use `p_gpair` and have the names of the loaded vectors, such as `svfloat64_t grad, svfloat64_t hess`, for readability?